From 8a428809dc0b40ec394d4e65e43eafa61b5828a8 Mon Sep 17 00:00:00 2001
From: Kevin Thorne <kevint@aptitekk.com>
Date: Mon, 7 Apr 2014 16:27:18 -0600
Subject: [PATCH 026/101] Added window rectangle for accurate mouse control,
 also added structure for the ability to update the WindowInfo

---
 .../Net/jrc/JRCServerApplication.java         | 17 ++++-
 .../Net/jrc/WindowControlAdapter.java         | 68 +++++++++++++++++++
 2 files changed, 84 insertions(+), 1 deletion(-)
 create mode 100644 TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/WindowControlAdapter.java

diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/JRCServerApplication.java b/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/JRCServerApplication.java
index bc40d80..40a88c7 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/JRCServerApplication.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/JRCServerApplication.java
@@ -29,6 +29,9 @@ public class JRCServerApplication extends Thread implements RemoteControlServerA
 	private RemoteControlServer server;
 	private boolean serverRunning = false;
 	
+	private WindowControlAdapter controlAdapter;
+	private ScreenRecorder screenRecorder;
+	
 	public JRCServerApplication(WindowInfo w) {
 		this.window = w;
 	}
@@ -91,7 +94,12 @@ public class JRCServerApplication extends Thread implements RemoteControlServerA
 
 	@Override
 	public ControlAdapter getControlAdapter() {
-		return new DesktopControlAdapter();
+		return this.controlAdapter;
+	}
+	
+	private ControlAdapter createControlAdapter() {
+		this.controlAdapter = new WindowControlAdapter(this.window);
+		return this.controlAdapter;
 	}
 
 	@Override
@@ -102,6 +110,8 @@ public class JRCServerApplication extends Thread implements RemoteControlServerA
 	
 	private void initialiseServer() throws Exception
 	{
+		createControlAdapter();
+		
 		String address = TeamScreenBroadcastThread.getLanIP();
 		List<String> slots = Arrays.asList(new String[]{"A","B","C","D","E","F"});
         Integer[] ports = new Integer[]{4455, 4456, 4457, 4458, 4459, 4460};
@@ -119,4 +129,9 @@ public class JRCServerApplication extends Thread implements RemoteControlServerA
 		return this.serverRunning;
 	}
 
+	public void updateWindowRect(WindowInfo info) {
+		this.window = info;
+		this.controlAdapter.updateWindow(info);
+	}
+
 }
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/WindowControlAdapter.java b/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/WindowControlAdapter.java
new file mode 100644
index 0000000..fa5143c
--- /dev/null
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/Net/jrc/WindowControlAdapter.java
@@ -0,0 +1,68 @@
+package com.aptitekk.TeamScreen.Net.jrc;
+
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+
+import org.one.stone.soup.remote.control.server.ControlAdapter;
+
+import com.aptitekk.TeamScreen.Util.WindowInfo;
+
+/**
+ * Controls commands for a specific window
+ *
+ * @author kevint.
+ *         Created Apr 7, 2014.
+ */
+public class WindowControlAdapter implements ControlAdapter {
+	
+	private Robot robot;
+	private WindowInfo window;
+	
+	public WindowControlAdapter(WindowInfo info)
+	{
+		this.window = info;
+		try{
+			this.robot = new Robot();
+		}
+		catch(Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+	
+	private Rectangle getWindowRect() {
+		Point topleft = new Point(this.window.getRect()[0], this.window.getRect()[1]);
+		Point bottomright = new Point(this.window.getRect()[2], this.window.getRect()[3]);
+		return new Rectangle(topleft.x, topleft.y, bottomright.x - topleft.x, bottomright.y - topleft.y);
+	}
+	
+	@Override
+	public void keyPress(int code) {
+		this.robot.keyPress(code);
+	}
+
+	@Override
+	public void keyRelease(int code) {
+		this.robot.keyRelease(code);
+	}
+
+	@Override
+	public void mouseMove(int x, int y) {
+		this.robot.mouseMove(getWindowRect().x+x,getWindowRect().y+y);
+	}
+
+	@Override
+	public void mousePress(int code) {
+		this.robot.mousePress(code);
+	}
+
+	@Override
+	public void mouseRelease(int code) {
+		this.robot.mouseRelease(code);
+	}
+	
+	public void updateWindow(WindowInfo info) {
+		this.window = info;
+	}
+}
-- 
2.39.5 (Apple Git-154)

